src/fu-engine.c

```diff
@@ -311,7 +311,8 @@ fu_engine_get_release_version (FuEngine *self, FuDevice *dev, XbNode *rel, GErro
 		return g_strdup (version);
 
 	/* don't touch my version! */
-	if (fmt == FWUPD_VERSION_FORMAT_PLAIN)
+	if (fmt == FWUPD_VERSION_FORMAT_PLAIN ||
+	    fmt == FWUPD_VERSION_FORMAT_HEX)
```

Comment:
Why doesn't version format hex parse as an integer properly?

Reply:
> 0x0003 gets transformed into 0x000003

Exactly, and then we do the dumb strcmp rather than a vercmp. I guess it's a UX thing; do we expect what the user sees in the terminal to match exactly what the metadinfo has? The purist in my would say yes, but if the hardware reports 0x0003 and it's `3` in the metainfo maybe that's just fine?

Reply:
I think treating plain and hex the same might be asking for issues on other stuff that uses hex.  What if the other devices with hex do it differently right now, now we stop being able to compare the actual values.

So I think I would rather just render hex as dropping leading zeros, but still treat as integer.  We can also simplify what `get-releases` shows to do the same thing.

Reply:
Hang on, back a bit, I'm insane. The chunk above this test actually tests the version using fu_common_vercmp_full() which will do the right thing for hex anyway -- in the event it fails we set an update error. I think we can just nuke the section completely!